File: src/frontend/types/connection.ts
export interface Connection {
  id: string;
  name: string;
  connection_string: string;
  type: 'standalone' | 'replica_set' | 'sharded';
  color: string | null;
  read_only: boolean;
  ssl_enabled: boolean;
  ssl_config: string | null;
  group_id: string | null;
  tags: string | null;
  created_at: string;
  updated_at: string;
}

export interface NewConnection {
  name: string;
  connection_string: string;
  type: 'standalone' | 'replica_set' | 'sharded';
  color: string | null;
  read_only: boolean;
  ssl_enabled: boolean;
  ssl_config: string | null;
  group_id: string | null;
  tags: string | null;
}

export interface UpdateConnection {
  name: string | null;
  connection_string: string | null;
  type: 'standalone' | 'replica_set' | 'sharded' | null;
  color: string | null;
  read_only: boolean | null;
  ssl_enabled: boolean | null;
  ssl_config: string | null;
  group_id: string | null;
  tags: string | null;
}

export interface ConnectionGroup {
  id: string;
  name: string;
  parent_id: string | null;
  sort_order: number;
  created_at: string;
}

export interface NewConnectionGroup {
  name: string;
  parent_id: string | null;
}

export interface ConnectionTestResult {
  success: boolean;
  message: string;
  databases: string[] | null;
}

export interface ConnectionFavorite {
  id: string;
  connection_id: string;
  sort_order: number;
  created_at: string;
}

File: src-tauri/src/models/connection.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Connection {
    pub id: String,
    pub name: String,
    pub connection_string: String,
    #[serde(rename = "type")]
    pub conn_type: String,
    pub color: Option<String>,
    pub read_only: bool,
    pub ssl_enabled: bool,
    pub ssl_config: Option<String>,
    pub group_id: Option<String>,
    pub tags: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct NewConnection {
    pub name: String,
    pub connection_string: String,
    #[serde(rename = "type")]
    pub conn_type: String,
    pub color: Option<String>,
    pub read_only: bool,
    pub ssl_enabled: bool,
    pub ssl_config: Option<String>,
    pub group_id: Option<String>,
    pub tags: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UpdateConnection {
    pub name: Option<String>,
    pub connection_string: Option<String>,
    #[serde(rename = "type")]
    pub conn_type: Option<String>,
    pub color: Option<String>,
    pub read_only: Option<bool>,
    pub ssl_enabled: Option<bool>,
    pub ssl_config: Option<String>,
    pub group_id: Option<String>,
    pub tags: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ConnectionGroup {
    pub id: String,
    pub name: String,
    pub parent_id: Option<String>,
    pub sort_order: i32,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct NewConnectionGroup {
    pub name: String,
    pub parent_id: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ConnectionTestResult {
    pub success: bool,
    pub message: String,
    pub databases: Option<Vec<String>>,
}

#[allow(dead_code)]
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ConnectionFavorite {
    pub id: String,
    pub connection_id: String,
    pub sort_order: i32,
    pub created_at: String,
}
